Sources/UtiliKit/Container/ContainerViewController.swift

```diff
@@ -85,67 +89,104 @@ private extension ContainerViewController {
+        if let interactor = interactor, interactor.wantsInteractiveStart {
+            interactor.startInteractiveTransition(context, using: animator)
+        } else {
+            animator.animateTransition(using: context)
+        }
     }
 }
 
 //MARK: Helper
 private extension ContainerViewController {
     
+    func configuredTransitionContext(from source: UIViewController, to destination: UIViewController) -> ContainerTransitionContext {
+        let context = ContainerTransitionContext(containerView: view, fromViewController: source, toViewController: destination)
+        context.completion = { [weak self] finished in
+            guard let self = self else { return }
+            self.finishTransitioning(from: source, to: destination, success: finished, animated: true)
+            self.delegate?.containerViewController(self, didFinishTransitioningFrom: source, to: destination)
+        }
+        
+        return context
+    }
+    
     func prepareForTransitioning(from source: UIViewController?, to destination: UIViewController, animated: Bool) {
         isTransitioning = true
         
         #if swift(>=4.2)
-        source?.willMove(toParent: nil)
-        destination.willMove(toParent: self)
+        source?.beginAppearanceTransition(false, animated: animated)
+        
+        destination.beginAppearanceTransition(true, animated: animated)
         addChild(destination)
+        destination.didMove(toParent: self)
         #else
-        source?.willMove(toParentViewController: nil)
-        destination.willMove(toParentViewController: self)
+        source?.beginAppearanceTransition(false, animated: animated)
+        
+        destination.beginAppearanceTransition(true, animated: animated)
         addChildViewController(destination)
+        destination.didMove(toParentViewController: self)
         #endif
     }
     
     func configure(destinationView: UIView, inContainer container: UIView) {
+        //It is the animator's responsibility (as with all UIViewControllerAnimatedTransitioning objects) to add the destinationView as a subview of the container view. The container will simply ensure the proper layout is used once the transition is completed.
+        
         destinationView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
         destinationView.frame = container.bounds
     }
     
-    func finishTransitioning(from source: UIViewController?, to destination: UIViewController, animated: Bool) {
-        source?.view.removeFromSuperview()
-        #if swift(>=4.2)
-        source?.removeFromParent()
-        source?.didMove(toParent: nil)
-        destination.didMove(toParent: self)
-        #else
-        source?.removeFromParentViewController()
-        source?.didMove(toParentViewController: nil)
-        destination.didMove(toParentViewController: self)
-        #endif
+    func finishTransitioning(from source: UIViewController?, to destination: UIViewController, success: Bool, animated: Bool) {
```

Comment:
Can you clean up the duplicate code contained in this function?
destination.endAppearanceTransition() is written 4 times for example

Reply:
Done. Also, now that we're supporting Swift 5 we should be ok to remove the pre- Swift 4.2 code.

Reply:
Great